### PR TITLE
feat(build): add Deno adapter

### PR DESCRIPTION
**Pull request by nakasyou**
I added adapter for `Deno` for build plugin.
It uses `Deno.serve`.
```ts
import denoBuildPlugin from '@hono/vite-build/deno'
```

### CODE REVIEW

File: packages/build/package.json [hunk 1]
```diff
@@ -33,8 +33,12 @@
       "import": "./dist/adapter/cloudflare-pages/index.js"
     },
     "./cloudflare-workers": {
-      "types": "./dist/adapter/cloudflare-workers/index.d.ts",
+      "types": "./dist/adapter/cloudflate-workers/index.d.ts",
```
Comment (yusukebe) [inline packages/build/package.json:36]: This is a typo. Should be:

```js
 "types": "./dist/adapter/cloudflare-workers/index.d.ts",
```

File: packages/build/src/adapter/deno/index.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            code += serveStaticHook(appName, {
+              filePaths: options?.staticPaths,
+              root: pluginOptions?.staticRoot,
+            })
+            return code
+          },
+        ],
+        entryContentAfterHooks: [
+          async (appName) => {
+            return `Deno.serve(${appName}.fetch)`
```
Comment (yusukebe) [inline packages/build/src/adapter/deno/index.ts:27]: Using `entryContentAfterHooks` is a good idea, but I think we don't have to use `Deno.serve`; the user can run it with the `deno serve` command without it. If we use `Deno.serve`, we can't specify the port number and other options with this place. You can make them options, but they will be verbose.

**Comment by yusukebe** [comment]
Hey @nakasyou 

Thank you for the great PR. I've left comments. The reason that the CI is failing is a typo of `./dist/adapter/cloudflate-workers/index.d.ts`.

File: packages/build/package.json (hunk 1 above)
Comment (nakasyou) [reply packages/build/package.json]: I fixed it.

File: packages/build/src/adapter/deno/index.ts (hunk 2 above)
Comment (nakasyou) [reply packages/build/src/adapter/deno/index.ts]: I agree. I changed.

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
Hi @nakasyou 

The CI is falling with another problem. The test became flaky. But this PR does not have a problem. So, I'll merge this first and fix it later. Thanks!

**Pull request merged**